examples/layer-surface.c

```diff
@@ -0,0 +1,434 @@
+};
+
+static void init_egl() {
+	EGLint config_attribs[] = {
+		EGL_SURFACE_TYPE, EGL_WINDOW_BIT,
+		EGL_RED_SIZE, 8,
+		EGL_GREEN_SIZE, 8,
+		EGL_BLUE_SIZE, 8,
+		EGL_RENDERABLE_TYPE, EGL_OPENGL_ES2_BIT,
+		EGL_NONE,
+	};
+
+	static const EGLint context_attribs[] = {
+		EGL_CONTEXT_CLIENT_VERSION, 2,
+		EGL_NONE,
+	};
+
+	egl_display = eglGetDisplay((EGLNativeDisplayType)display);
+	if (egl_display == EGL_NO_DISPLAY) {
+		fprintf(stderr, "Can't create egl display\n");
+		exit(EXIT_FAILURE);
+	} else {
+		fprintf(stderr, "Created egl display\n");
+	}
+
+	EGLint major, minor;
+	if (eglInitialize(egl_display, &major, &minor) != EGL_TRUE) {
+		fprintf(stderr, "Can't initialise egl display\n");
+		exit(EXIT_FAILURE);
+	}
+	printf("EGL major: %d, minor %d\n", major, minor);
+
+	if (!eglBindAPI(EGL_OPENGL_ES_API)) {
+		fprintf(stderr, "Can't bind API\n");
+		exit(EXIT_FAILURE);
+	} else {
+		fprintf(stderr, "Bound API\n");
+	}
+
+	EGLint count;
+	eglGetConfigs(egl_display, NULL, 0, &count);
+	printf("EGL has %d configs\n", count);
+
+	EGLConfig *configs = calloc(count, sizeof(EGLConfig));
+	if (configs == NULL) {
+		fprintf(stderr, "Can't allocate EGL configs\n");
+		exit(EXIT_FAILURE);
+	}
+
+	EGLint n;
+	eglChooseConfig(egl_display, config_attribs, configs, count, &n);
+
+	EGLint size;
+	for (int i = 0; i < n; ++i) {
+		eglGetConfigAttrib(egl_display, configs[i], EGL_BUFFER_SIZE, &size);
+		printf("Buffer size for config %d is %d\n", i, size);
+		eglGetConfigAttrib(egl_display, configs[i], EGL_RED_SIZE, &size);
+		printf("Red size for config %d is %d\n", i, size);
+
+		egl_conf = configs[i];
+		break;
+	}
+
+	egl_context = eglCreateContext(egl_display, egl_conf, EGL_NO_CONTEXT,
+		context_attribs);
+}
+
+static void create_window() {
+	egl_window = wl_egl_window_create(surface, width, height);
+	if (egl_window == EGL_NO_SURFACE) {
+		fprintf(stderr, "Can't create egl window\n");
+		exit(EXIT_FAILURE);
+	} else {
+		fprintf(stderr, "Created egl window\n");
+	}
+
+	egl_surface = eglCreateWindowSurface(egl_display, egl_conf,
+		(EGLNativeWindowType)egl_window, NULL);
+
+	draw();
+
+	wl_display_dispatch(display);
+	wl_display_roundtrip(display);
+}
+
+enum surface_layers_layer parse_layer(const char *s) {
+	if (strcmp(s, "background") == 0) {
+		return SURFACE_LAYERS_LAYER_BACKGROUND;
+	} else if (strcmp(s, "bottom") == 0) {
+		return SURFACE_LAYERS_LAYER_BOTTOM;
+	} else if (strcmp(s, "top") == 0) {
+		return SURFACE_LAYERS_LAYER_TOP;
+	} else if (strcmp(s, "overlay") == 0) {
+		return SURFACE_LAYERS_LAYER_OVERLAY;
+	}
+	fprintf(stderr, "Unknown layer name: %s\n", s);
+	exit(EXIT_FAILURE);
+}
+
+enum layer_surface_anchor parse_anchor(const char *s) {
+	if (strcmp(s, "top") == 0) {
+		return LAYER_SURFACE_ANCHOR_TOP;
+	} else if (strcmp(s, "bottom") == 0) {
+		return LAYER_SURFACE_ANCHOR_BOTTOM;
+	} else if (strcmp(s, "left") == 0) {
+		return LAYER_SURFACE_ANCHOR_LEFT;
+	} else if (strcmp(s, "right") == 0) {
+		return LAYER_SURFACE_ANCHOR_RIGHT;
+	}
```

Comment:
This is a bitmask - you can be top-right, bottom-right, etc.

Reply:
can you be top-bottom or right-left?

Reply:
Yes, just specify `-a` multiple times (e.g. `-a top -a left`).